src/main/java/org/opensearch/knn/training/TrainingJob.java

```diff
@@ -182,6 +187,14 @@ public void run() {
                 KNNSettings.state().getSettingValue(KNNSettings.KNN_ALGO_PARAM_INDEX_THREAD_QTY)
             );
 
+            if (VectorDataType.BINARY.equals(model.getModelMetadata().getVectorDataType())) {
+                trainParameters.put(
+                    KNNConstants.INDEX_DESCRIPTION_PARAMETER,
+                    FAISS_BINARY_INDEX_DESCRIPTION_PREFIX + trainParameters.get(KNNConstants.INDEX_DESCRIPTION_PARAMETER).toString()
```

Comment:
I see this happening in a couple places. We should abstract into a single method.

Reply:
abstracted into a single method.